### ISSUE TITLE
Desktop client gets stuck in loop when trying to authenticate on Linux

### ISSUE THREAD

**Issue by danbohea** [nextcloud/desktop#1192]
### Expected behaviour
The desktop client authenticates with the remote Nextcloud instance and I progress to whatever the next stage of the wizard is.

### Actual behaviour
I get stuck in a loop of being asked to authenticate.

### Steps to reproduce
1. Install client via official PPA and launch the client
2. Enter the URL of my remote Nextcloud instance.
3. I took screen grabs to illustrate the next steps:

![02](https://user-images.githubusercontent.com/6885065/55718807-93ea2a00-59f4-11e9-9ddc-f6d2bd56b324.png)
![03](https://user-images.githubusercontent.com/6885065/55718812-964c8400-59f4-11e9-8433-e1724f571d3c.png)
![04](https://user-images.githubusercontent.com/6885065/55718815-98164780-59f4-11e9-9fab-21c951d4187d.png)
![05](https://user-images.githubusercontent.com/6885065/55718820-99e00b00-59f4-11e9-8675-ce209d13f8d0.png)

I then get stuck in a loop of the last 3 screens.

- I've tried authenticating via generated key but it seems to make no difference, I still get the same loop.
- I've successfully connected via webdav.
- I've successfully connected via Android app.
- I've tried the appimage instead of the apt package and I get the same issue.
- I've tried to set up the client on another system (same OS) and I get the same issue.
- My Nextcloud instance is behind a firewall but TCP ports 80 & 443 are open for business.

### Client configuration
Client version:
2.5.2git

Operating system:
KDE Neon 5.15

OS language:
?

Qt version used by client package (Linux only, see also Settings dialog):
I don't see this.

Client package (From Nextcloud or distro) (Linux only):
`nextcloud-client` via `ppa:nextcloud-devs/client`

Installation path of client:
```sh
$ whereis nextcloud
nextcloud: /usr/bin/nextcloud /usr/lib/x86_64-linux-gnu/nextcloud /usr/share/nextcloud
```

### Server configuration

Installed via `snap`.

Operating system: Ubuntu 18.04

Nextcloud version: 15.0.5

**Comment by StefanBCN** [comment]
I get the same problem on the OSX release (2.5.2 via `brew cask`, server version is 15.0.6 on Ubuntu 16.04)

**Comment by danbohea** [comment]
Now fixed by setting `overwriteprotocol` to `https` on the remote Nextcloud instance:

```sh
sudo nextcloud.occ config:system:set overwriteprotocol --value="https"
```

For anyone following this, I'm prefixing `occ` with `nextcloud.` specifically because I installed nextcloud via `snap`.

In my case, SSL is being handled by Cloudflare so Nextcloud was unaware that it was being used. At least, that's what I've gathered by reading https://help.nextcloud.com/t/stuck-in-a-login-loop-when-logging-in-from-sync-client-on-linux/47549/6

I would be great if the desktop client provided more feedback as to why it's not connecting successfully rather than saying nothing and looping the login procedure.
